### PR TITLE
feat(charts): Add auto-generated aria labels

### PR DESCRIPTION
**Pull request by vuluongj20**
Create a util called `useAria` to automatically generate aria-labels based on the data provided.

Screen Shot 2022-10-04 at 4 42 54 PM

**Here's a sample label:**
> "line chart covering Oct 3, 4:30 PM to Oct 4, 4:30 PM featuring 2 data series: tpm() and previous tpm(). The tpm() series contains 97 data points. Its lowest value is 6935.533 on Oct 4, 4:30 PM and highest value is 18276.933 on Oct 4, 4:30 PM. The previous tpm() series contains 96 data points. Its lowest value is 4911.333 achieved on Oct 3, 4:30 PM and highest value is 18506.2 achieved on Oct 3, 4:30 PM"

ECharts itself provides auto-generated labels (disabled by default). However, imo it's not very good and does not suit our needs ([docs](https://echarts.apache.org/handbook/en/best-practices/aria/#)). **Here's what ECharts would generate for the same chart:**
> "This is a chart. It consists of 2 series count. The 0 series is a Line chart representing tpm().The first 10 items are: 1664840700000, 8522.6, 1664841600000, 9144, 1664842500000, 8786.866666666667, 1664843400000, 8531.533333333333, 1664844300000, 9283.266666666666, 1664845200000, 10109.266666666666, 1664846100000, 10033.8, 1664847000000, 10372.666666666666, 1664847900000, 9167, 1664848800000, 9432.466666666667.  The 1 series is a Line chart representing previous tpm().The first 10 items are: 1664840700000, 5439.2, 5439.2, NaN, 1664841600000, 5407, 5407, NaN, 1664842500000, 5547.266666666666, 5547.266666666666, NaN, 1664843400000, 6253.4, 6253.4, NaN, 1664844300000, 6710.8, 6710.8, NaN, 1664845200000, 7349.8, 7349.8, NaN, 1664846100000, 7519, 7519, NaN, 1664847000000, 7368.666666666667, 7368.666666666667, NaN, 1664847900000, 7129.866666666667, 7129.866666666667, NaN, 1664848800000, 7258.4, 7258.4, NaN."

### CODE REVIEW

File: static/app/components/charts/baseChart.tsx [hunk 1]
```diff
@@ -67,6 +68,8 @@ const handleClick = (clickSeries: any, instance: ECharts) => {
   }
 };
 
+echarts.use(AriaComponent);
```
Comment (malwilley) [inline static/app/components/charts/baseChart.tsx:71]: Looks like `echarts` is mocked by `tests/js/sentry-test/echartsMock.js` so this is failing in tests. Will have to add a `use` method it looks like?

File: static/app/components/charts/utils.tsx [hunk 2]
```diff
@@ -279,3 +285,98 @@ export const processTableResults = (tableResults?: TableDataWithTitle[]) => {
...
 };
+
+function formatList(items: Array<string | number | undefined>) {
+  const filteredItems = items.filter(item => !!item);
+  return [[...filteredItems].slice(0, -1).join(', '), [...filteredItems].slice(-1)]
+    .filter(type => !!type)
+    .join(' and ');
+}
+
+export function useAria(
```
Comment (malwilley) [inline static/app/components/charts/utils.tsx:296]: Nit: could have a more descriptive name like `useEchartsAriaLabels` or something

**Review by malwilley — CHANGES_REQUESTED** [review]
This is a great accessibility improvement! Thanks for the detailed PR description too, very helpful 😄

File: static/app/components/charts/utils.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/components/charts/utils.tsx:296]: Not even a nit, definitely worth changing!

**Comment by evanpurkhiser** [comment]
This is AWESOME

File: static/app/components/charts/utils.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/components/charts/utils.tsx]: Ahh great point! Renamed in https://github.com/getsentry/sentry/pull/39653/commits/4076bf61bdf4fe218e4e9c50858bdae4c7d58bb0

File: static/app/components/charts/baseChart.tsx (hunk 1 above)
Comment (vuluongj20) [reply static/app/components/charts/baseChart.tsx:76]: Ohhhh that's how you do it. Thanks so much! The tests should be passing now – https://github.com/getsentry/sentry/pull/39653/commits/45957c65b195837812e2a0b9455a6952643d5ef9

**Review by evanpurkhiser — APPROVED** [review]

**Review by malwilley — APPROVED** [review]

**Pull request merged**